pkg/services/sqlstore/migrations/accesscontrol/datasource_drilldown_removal.go

```diff
@@ -0,0 +1,39 @@
+package accesscontrol
+
+import (
+	"fmt"
+
+	"xorm.io/xorm"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+const (
+	datasourceDrilldownRemoval = "delete all drilldown actions, their assignments, and related roles"
+)
+
+func AddDatasourceDrilldownRemovalMigration(mg *migrator.Migrator) {
+	mg.AddMigration(datasourceDrilldownRemoval, &datasourceDrilldownRemovalMigrator{})
+}
+
+type datasourceDrilldownRemovalMigrator struct {
+	migrator.MigrationBase
+}
+
+func (m *datasourceDrilldownRemovalMigrator) SQL(dialect migrator.Dialect) string {
+	return CodeMigrationSQL
+}
+
+func (m *datasourceDrilldownRemovalMigrator) Exec(sess *xorm.Session, mg *migrator.Migrator) error {
+	result, err := sess.Exec("DELETE FROM permission WHERE action = ?", "datasources:drilldown")
+	if err != nil {
+		return err
+	}
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		return err
+	}
+	mg.Logger.Info(fmt.Sprintf("Removed %d datasources:drilldown permissions", rowsAffected))
```

Comment:
Imo we should not error out if we have an issue knowing how many rows have been affected. But that's an opinion only.
```suggestion
	if rowsAffected, err := result.RowsAffected(); err == nil {
		mg.Logger.Info(fmt.Sprintf("Removed %d datasources:drilldown permissions", rowsAffected))
	}
```